### PR TITLE
Fix all -Wtype-limits warnings.

### PR DESCRIPTION
**Pull request by marxin**

### CODE REVIEW

File: platform/x11/os_x11.cpp [hunk 1]
```diff
@@ -1743,7 +1743,7 @@ void OS_X11::handle_key_event(XKeyEvent *p_event, bool p_echo) {
 			// is correct, but the xorg developers are
 			// not very helpful today.
 
-			::Time tresh = ABS(peek_event.xkey.time - xkeyevent->time);
+			::Time tresh = peek_event.xkey.time - xkeyevent->time;
```
Comment (akien-mga) [inline platform/x11/os_x11.cpp]: Are we sure that `peek_event.xkey.time > xkeyevent->time` at all times?

**Comment by akien-mga** [comment]
Needs a rebase after the merge of #25853.

File: platform/x11/os_x11.cpp (hunk 1 above)
Comment (Dar13) [reply platform/x11/os_x11.cpp]: The original `ABS` macro wasn't quite correct due to the types involved being unsigned. The `ABSDIFF` macro defined in `servers/audio/reverb_sw.cpp` does the correct thing for unsigned integers. Perhaps move that macro into `core/typedefs.h` alongside `ABS` and use it here?

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**